### PR TITLE
Add Route Security to some routes

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves: #487 

## Description

Add additional security to routes to make sure they are requesting resources for them in the namespaces they expect to have.

## How Has This Been Tested?

See ticket

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lucferbux — APPROVED** [review]
/lgtm

Tested it and it seems to be working as expected, only retrieving objects from the dashboard or notebook namespace

**Review by samuelvl — APPROVED** [review]
/lgtm

File: backend/src/utils/route-security.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const { notebookNamespace, dashboardNamespace } = getNamespaces(fastify);
+  const username = await getUserName(fastify, request);
+  const translatedUsername = usernameTranslate(username);
+
+  if (dashboardNamespace === notebookNamespace || namespace === dashboardNamespace) {
+    // Requesting something from the exclusive dashboard namespace -- can't validate cleanly
+    return;
+  }
+
+  if (namespace !== notebookNamespace) {
```
Comment (kpouget) [inline backend/src/utils/route-security.ts:51]: minor, but I find these guards hard to read.
Could be something like:
```
  if (namespace === dashboardNamespace) {
    // Requesting something from the dashboard namespace -- can't validate cleanly
    return;
  }

  if (namespace !== notebookNamespace) {
    ...
    throw createCustomError(
      'Wrong namespace',
      'Cannot request a resource outside of the notebook namespace',
      403,
    );
  }

  if (!name.includes(translatedUsername)) {
    ...
  }
  
return // namespace and name are valid, authorize the request # have this explicit here
```

File: backend/src/utils/route-security.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    fastify.log.error(
+      `Tried to interact with a resource outside (${namespace}) of the notebook namespace (${dashboardNamespace})`,
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply backend/src/utils/route-security.ts:53]: Yeah, I was testing and went back and forth on that one -- good catch. The error message will print the wrong namespace 🤦 Minor but probably something I should fix.

File: backend/src/utils/route-security.ts (hunk 1 above)
Comment (andrewballantyne) [reply backend/src/utils/route-security.ts:51]: I don't follow your request -- is the only thing you added a comment at the bottom with an empty return?

Comment (kpouget) [reply backend/src/utils/route-security.ts:51]: yeah that's minor, maybe just bikeshedding, but I find it hard to read these two tests:
```
  if (dashboardNamespace === notebookNamespace || namespace === dashboardNamespace) {...}
  if (namespace !== notebookNamespace) {...}
```
and I don't really understand which case is supposed to reach the end of the function

**Pull request merged**